ospf6d/ospf6_interface.c

```diff
@@ -1008,6 +1008,108 @@ DEFUN (show_ipv6_ospf6_interface,
 	return CMD_SUCCESS;
 }
 
+static int ospf6_interface_show_traffic(struct vty *vty,
+					uint32_t vrf_id,
+					char *intf_name,
+					int display_once)
+{
+	struct interface *ifp;
+	struct vrf *vrf = NULL;
+	struct ospf6_interface *oi;
+
+	vrf = vrf_lookup_by_id(vrf_id);
+
+	if (!display_once) {
+		vty_out(vty, "\n");
+		vty_out(vty, "%-12s%-17s%-17s%-17s%-17s%-17s\n",
+			"Interface", "    HELLO", "    DB-Desc", "   LS-Req",
+			"   LS-Update", "   LS-Ack");
+		vty_out(vty, "%-10s%-18s%-18s%-17s%-17s%-17s\n", "",
+			"      Rx/Tx", "     Rx/Tx", "    Rx/Tx", "    Rx/Tx", "    Rx/Tx");
+		vty_out(vty,
+		"--------------------------------------------------------------------------------------------\n");
+	}
+
+	if (intf_name == NULL) {
+		FOR_ALL_INTERFACES (vrf, ifp) {
+			if (ifp->info)
+				oi = (struct ospf6_interface *)ifp->info;
+			else
+				continue;
+
+			vty_out(vty,
+			"%-10s %8u/%-8u %7u/%-7u %7u/%-7u %7u/%-7u %7u/%-7u\n",
+				oi->interface->name, oi->hello_in,
+				oi->hello_out,
+				oi->db_desc_in, oi->db_desc_out,
+				oi->ls_req_in, oi->ls_req_out,
+				oi->ls_upd_in, oi->ls_upd_out,
+				oi->ls_ack_in, oi->ls_ack_out);
+		}
+	} else {
+		ifp = if_lookup_by_name(intf_name, vrf_id);
+		if (ifp == NULL) {
+			vty_out(vty,
+				"No such Interface: %s\n",
+				intf_name);
+			return 0;
+		} else if (ifp->info == NULL) {
+			vty_out(vty,
+				"   OSPF not enabled on this interface %s\n",
+				intf_name);
+			return 0;
+		}
+
+		oi = (struct ospf6_interface *)ifp->info;
+
+		vty_out(vty,
+			"%-10s %8u/%-8u %7u/%-7u %7u/%-7u %7u/%-7u %7u/%-7u\n",
+			oi->interface->name, oi->hello_in,
+			oi->hello_out,
+			oi->db_desc_in, oi->db_desc_out,
+			oi->ls_req_in, oi->ls_req_out,
+			oi->ls_upd_in, oi->ls_upd_out,
+			oi->ls_ack_in, oi->ls_ack_out);
+	}
+
+	return CMD_SUCCESS;
+}
+
+/* show interface */
+DEFUN (show_ipv6_ospf6_interface_traffic,
+       show_ipv6_ospf6_interface_traffic_cmd,
+       "show ipv6 ospf6 interface traffic [IFNAME]",
+       SHOW_STR
+       IP6_STR
+       OSPF6_STR
+       INTERFACE_STR
+       "Protocol Packet counters\n"
+       IFNAME_STR)
+{
+	int idx_ifname = 0;
+	int display_once = 0;
+	char *intf_name = NULL;
+	struct interface *ifp = NULL;
+
+	if (argv_find(argv, argc, "IFNAME", &idx_ifname)) {
+		intf_name = argv[idx_ifname]->arg;
+		ifp = if_lookup_by_name(intf_name, VRF_DEFAULT);
+		if (ifp == NULL) {
+			vty_out(vty,
+				"No such Interface: %s\n",
+				intf_name);
+			return CMD_WARNING;
+		}
+	}
+
+	ospf6_interface_show_traffic(vty, VRF_DEFAULT, intf_name,
```

Comment:
I'd prefer to pass `ifp` instead of `intf_name` to avoid having to call `if_lookup_by_name()` again later. Also, I see that `display_once` is always set to zero here. Do we really need this variable?